implementations/rust/ockam/ockam_api/src/cloud/share/create.rs

```diff
@@ -0,0 +1,163 @@
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub grant_role: RoleInShare,
+    #[n(3)] pub recipient_email: Option<String>,
+    #[n(4)] pub remaining_uses: Option<usize>,
+    #[n(5)] pub scope: ShareScope,
+    #[n(6)] pub target_id: String,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateServiceInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub project_id: String,
+    #[n(3)] pub recipient_email: String,
+
+    // TODO: Is there an identity type I should use that may be CBOR-encoded?
+    // TODO: Should route be a MultiAddr?
+    #[n(4)] pub project_identity: IdentityIdentifier,
+    #[n(5)] pub project_route: String,
+    #[n(6)] pub project_authority_identity: IdentityIdentifier,
+    #[n(7)] pub project_authority_route: String,
+    #[n(8)] pub shared_node_identity: IdentityIdentifier,
+    #[n(9)] pub shared_node_route: String,
+}
+
+mod node {
+    use ockam_core::api::{Request, Response};
+    use ockam_core::{self, Result};
+    use ockam_multiaddr::MultiAddr;
+    use ockam_node::Context;
+    use tracing::trace;
+
+    use crate::cloud::CloudRequestWrapper;
+    use crate::nodes::{NodeManager, NodeManagerWorker};
+
+    use super::*;
+
+    const API_SERVICE: &str = "users";
+
+    impl NodeManager {
+        // TODO: Should identity_name not be an Option for this use-case?
+        //       This is an authenticated-only request
+        pub async fn create_invite(
+            &self,
+            ctx: &Context,
+            req: CreateInvite,
+            route: &MultiAddr,
+            identity_name: Option<String>,
+        ) -> Result<SentInvitation> {
+            Response::parse_response_body(
+                self.create_invite_response(
+                    ctx,
+                    CloudRequestWrapper::new(req, route, identity_name),
+                )
+                .await?
+                .as_slice(),
+            )
+        }
+
+        pub(crate) async fn create_invite_response(
+            &self,
+            ctx: &Context,
+            req_wrapper: CloudRequestWrapper<CreateInvite>,
+        ) -> Result<Vec<u8>> {
+            let cloud_multiaddr = req_wrapper.multiaddr()?;
+            let req_body = req_wrapper.req;
+
+            let label = "create_invite";
+            trace!(%req_body.scope, target_id = %req_body.target_id, "creating invite");
+
+            let req_builder = Request::post("/v0/invites").body(req_body);
+
+            self.request_controller(
+                ctx,
+                label,
+                "create_invite",
+                &cloud_multiaddr,
+                API_SERVICE,
+                req_builder,
+                None,
+            )
+            .await
+        }
+
+        // TODO: Should identity_name not be an Option for this use-case?
+        //       This is an authenticated-only request
```

Comment:
This is related to https://github.com/build-trust/ockam/pull/5451/files#r1277582610. I think it still makes sense to have the option, in case we want to use different project/identities pairs from the same machine.